pytensor/tensor/rewriting/elemwise.py

```diff
@@ -1025,3 +1027,41 @@ def local_careduce_fusion(fgraph, node):
     "fusion",
     position=49,
 )
+
+
+@node_rewriter([Elemwise])
+def inline_elemwise_scan(fgraph, node):
+    from pytensor.scan.basic import scan
+
+    scalar_op = node.op.scalar_op
+
+    if not isinstance(scalar_op, ScalarScanOp):
+        return None
+
+    # TODO: Add non-batched implementation?
+    n_carried_inputs = scalar_op.nout - scalar_op.nin
+    bcasted_inputs = broadcast_arrays(*node.inputs)
+    ret, updates = scan(
```

Comment:
Instead of returning a vanilla scan we could also return the equivalent output from `save_mem_new_scan`, just a lot of duplicated work?